src/lib/sw.js

```diff
@@ -187,16 +189,14 @@ self.addEventListener("fetch", (event) => {
   event.respondWith(p);
 });
 
-workbox.routing.setCatchHandler(async ({event}) => {
+workboxRouting.setCatchHandler(async ({event}) => {
   // Destination is set by loading this content normally; it's not set for fetch(), so look for our
   // custom header.
   const isDocumentRequest =
     event.request.destination === "document" ||
     event.request.headers.get("X-Document");
   if (isDocumentRequest) {
-    const cacheKey = workbox.precaching.getCacheKeyForURL(
-      "/offline/index.html",
-    );
+    const cacheKey = workboxPrecaching.getCacheKeyForURL("/offline/index.html");
```

Comment:
You might be able to use https://developers.google.com/web/tools/workbox/guides/migrations/migrate-from-v4#reading_precached_responses here as well.

Reply:
This doesn't work as we're not _precaching_ offline in development. We're just checking the runtime cache—it looks like `matchPrecache` does what it says on the tin, only matches precache, not runtime :)

Reply:
Sorry, I'm not clear what you mean here—I commented on a line that was previously calling `workbox.precaching.getCacheKeyForURL()`, which leads me to believe that you were checking for a precached entry, not a runtime cached entry?

If your point was that it might be a cache miss, that's okay—`matchPrecache()` will just return `null` in that scenario, and you can check for that.